pkg/options/types_test.go

```diff
@@ -162,13 +162,25 @@ func TestNodeFieldSelector(t *testing.T) {
 		Wanted string
 	}{
 		{
```

Comment:
Add a test that it can differentiate between nil and "".

Reply:
> ```
> `spec.nodeName=""`  # ""
> `spec.nodeName=` # nil
> ```
> 
> This might work, since we can't use " spec.nodeName="" "

I thought so too, so I added regulars to prevent this, as well as modifications to the configuration.

Reply:
> ```
> `spec.nodeName=""`  # ""
> `spec.nodeName=` # nil
> ```
> 
> This might work, since we can't use " spec.nodeName="" "

I thought so too, so I added regulars to prevent this, as well as modifications to the configuration.

Reply:
> Add a test that it can differentiate between nil and "".

I fixed it locally. Submit it later along with other questions
<img width="775" alt="image" src="https://github.com/kubernetes/kube-state-metrics/assets/1990362/cb2f6196-5e10-4577-9c40-9173e7cfb996">